lib/fog/storage/google_json/utils.rb

```diff
@@ -19,7 +19,7 @@ def url(params, expires)
 
         def host_path_query(params, expires)
           params[:headers]["Date"] = expires.to_i
-          params[:path] = URI.encode(params[:path]).gsub("%2F", "/")
+          params[:path] = CGI.escape(params[:path]).gsub("%2F", "/").gsub("+", "%20")
```

Comment:
As described in https://stackoverflow.com/a/13059657/1992201, I believe `CGI.escape` is for escaping a URL value in the query string. https://github.com/fog/fog-google/pull/517 was my approach.

Reply:
```ruby
require 'cgi'
test = "some/string with space & text"
CGI.escape(test).gsub("%2F", "/").gsub("+", "%20")
=> "some/string%20with%20space%20%26%20text"
::Addressable::URI.encode_component(test, Addressable::URI::CharacterClasses::PATH).gsub("%2F", "/")
=> "some/string%20with%20space%20&%20text"
```

We do see here the `&` is not escaped in the latter case, but it is in the first. Do we care?